swan-lake/resources/featured-scenarios/write-a-restful-api-with-ballerina.md

```diff
@@ -127,103 +187,75 @@ service /covid/status on new http:Listener(9000) {
 In this code:
 
 - Unlike normal functions, resource methods can have accessors. In this case, the accessor is set to `get`, which means only HTTP `GET` requests could hit this resource. Ballerina automatically serializes Ballerina records as JSON and sends them over the wire. 
-- The default HTTP response status code for a resource method other than `post` is `200 OK`. For an HTTP `POST` resource, the default HTTP response status code is `201 Created`. 
+- The default HTTP response status code for a resource method other than `post` is `200 OK`. The default HTTP response status code for an HTTP `POST` resource is `201 Created`. 
 
 ### Create the second resource to add data
 
-To create the second resource of the first endpoint to add new COVID-19 data to the dataset by ISO code, add the code below to the API template file (i.e., `service.bal`).
+Create the second resource of the first endpoint to add new COVID-19 data to the dataset by ISO code, using the [Ballerina HTTP API Designer](/learn/vs-code-extension/design-the-services/http-api-designer/) in VS Code, as shown below.
+
+>**Note:** Define an HTTP resource that allows the `POST` operation on the resource path `countries` and accepts a `CovidEntry[]` type payload named `covidEntries`. Use `CovidEntry[]` and `ConflictingIsoCodesError` as the response types.
 
+![Create POST resource](/learn/images/featured-scenarios/write-a-restful-api-with-ballerina/create-post-resource.gif)
+
+Implement the logic of this `POST` resource function with the code below.
 
 ```ballerina
-resource function post countries(@http:Payload CovidEntry[] covidEntries)
-                                    returns CovidEntry[]|ConflictingIsoCodesError {
+resource function post countries(CovidEntry[] covidEntries)
+                                returns CovidEntry[]|ConflictingIsoCodesError {
 
     string[] conflictingISOs = from CovidEntry covidEntry in covidEntries
         where covidTable.hasKey(covidEntry.iso_code)
         select covidEntry.iso_code;
 
     if conflictingISOs.length() > 0 {
         return {
-            body: {
-                errmsg: string:'join(" ", "Conflicting ISO Codes:", ...conflictingISOs)
-            }
+            body: string:'join(" ", "Conflicting ISO Codes:", ...conflictingISOs)
         };
     } else {
```

Comment:
The `else` part is redundant here right?

Reply:
Removed.